artifactory/utils/docker/docker.go

```diff
@@ -314,3 +322,43 @@ func DockerLogin(imageTag string, config *DockerLoginConfig) error {
 	// Login succeeded
 	return nil
 }
+
+// Version command
+type versionCmd struct{}
+
+func (versionCmd *versionCmd) GetCmd() *exec.Cmd {
+	if cliutils.IsWindows() {
+		return exec.Command("cmd", "/C", "echo", "docker --version")
+	}
+	return exec.Command("sh", "-c", "docker --version")
+}
+
+func (versionCmd *versionCmd) GetEnv() map[string]string {
+	return map[string]string{}
+}
+
+func (versionCmd *versionCmd) GetStdWriter() io.WriteCloser {
+	return nil
+}
+
+func (versionCmd *versionCmd) GetErrWriter() io.WriteCloser {
+	return nil
+}
+
+func IsVersionSupported() error {
```

Comment:
Functions with the *is* prefix usually return a boolean.
Therefore, either have this function return a boolean, or rename it to *validateVersion*.
I think that it is better to return a boolean, so that errors that is caused due to execution failure are distinguishable than unsupported versions.

Reply:
Done